crates/oxc_hir/src/hir_util.rs

```diff
@@ -0,0 +1,75 @@
+use crate::hir::{
+    ArrayExpressionElement, Expression, ObjectProperty, ObjectPropertyKind, PropertyKey,
+    SpreadElement,
+};
+
+/// Returns true if this is a literal value. We define a literal value as any node that evaluates
+/// to the same thing regardless of when or where it is evaluated. So `/xyz/` and `[3, 5]` are
+/// literals, but the name a is not.
+///
+/// Function literals do not meet this definition, because they lexically capture variables. For
+/// example, if you have `function() { return a; }`.
+/// If it is evaluated in a different scope, then it captures a different variable. Even if
+/// the function did not read any captured variables directly, it would still fail this definition,
+/// because it affects the lifecycle of variables in the enclosing scope.
+///
+/// However, a function literal with respect to a particular scope is a literal.
+pub trait IsLiteralValue<'a, 'b> {
```

Comment:
It would be appreciated if we can link the source. For anything that's copied over, I want the original source reference so people know where this is from.

Reply:
@mysteryven Just a single small change here.

Reply:
Done.